packages/shared/src/sentry.ts

```diff
@@ -0,0 +1,114 @@
+import * as Sentry from "@sentry/node";
+const { nodeProfilingIntegration } = require("@sentry/profiling-node");
+import { AuthInfo } from "./graphql/fetchAuthInfoFromGraphQL";
+import { getDeviceId } from "./getDeviceId";
+import { randomUUID } from "crypto";
+import { logger } from "./logger";
+import { formatTags, Tags } from "./formatTags";
+
+const SENTRY_DSN =
+  "https://5c145b72bb502832982243d6584f163d@o437061.ingest.us.sentry.io/4507534763819008"; // write-only permissions
+
+class SentryAPI {
+  private deviceId: string;
+  private sessionId: string;
+  private authInfo?: AuthInfo;
+  private initialized: boolean = false;
+
+  constructor() {
+    this.deviceId = getDeviceId();
+    this.sessionId = randomUUID();
+  }
+
+  initialize(app: string, version: string | undefined) {
+    if (this.initialized) {
+      console.warn(`Logger already initialized.`);
+    }
+
+    Sentry.init({
+      dsn: SENTRY_DSN,
+      integrations: [nodeProfilingIntegration()],
+    });
+
+    Sentry.setTags({
+      app,
+      version,
+    });
+
+    this.initialized = true;
+  }
+
+  identify(authInfo: AuthInfo) {
+    this.authInfo = authInfo;
+  }
+
+  async close() {
+    if (process.env.REPLAY_TELEMETRY_DISABLED) {
+      return;
+    }
+
+    await Sentry.close();
+  }
+
+  captureException(error: Error, tags?: Tags) {
+    if (process.env.REPLAY_TELEMETRY_DISABLED) {
+      return;
+    }
+
+    const formattedTags = formatTags(tags);
+
+    const entry: Record<string, any> = {
+      ...formattedTags,
+      deviceId: this.deviceId,
+      sessionId: this.sessionId,
+    };
+
+    if (this.authInfo) {
+      switch (this.authInfo.type) {
+        case "user": {
+          entry.userId = this.authInfo.id;
+          break;
+        }
+        case "workspace": {
+          entry.workspaceId = this.authInfo.id;
+          break;
+        }
+      }
+    }
+
+    Sentry.captureException(error, entry);
+  }
+}
+
+export const sentry = new SentryAPI();
+
+// This should be called with the name once at the entry point.
+// For example, with the Playwright plugin, it is called in the Reporter interface constructor.
+export function initSentry(app: string, version: string | undefined) {
+  sentry.initialize(app, version);
+}
+
+export async function withSentry<T>(
+  methodName: string,
+  fn: () => T | Promise<T>,
+  tags?: Tags
+): Promise<T> {
+  try {
+    const result = await fn();
+    return result;
+  } catch (error: any) {
+    logger.error(`${methodName}Failed`, tags);
```

Comment:
I feel like this one might be too hidden here. It lacks symmetry with `*:Started` logs and similar. When reading code sites that use this I'd be tempted to add `logger.error` to *them* because I wouldn't realize that this is already auto-handled within this function.

We could add `*:Started` logs here but I think it would still be surprising. It doesn't like `withSentry` should step outside of its core responsibility (integrating with Sentry).